src/core/packages/saved-objects/api-server-internal/src/lib/apis/bulk_create.ts

```diff
@@ -180,9 +186,35 @@ export const performBulkCreate = async <T>(
     preflightCheckObjects
   );
 
+  const accessControlPreflightObjects = validObjects
+    .filter(({ value }) => value.accessControlPreflightIndex !== undefined)
+    .map(({ value }) => {
+      const { type, id, initialNamespaces } = value.object;
+      const namespaces = initialNamespaces ?? [namespaceString];
+      return { type, id, overwrite, namespaces };
+    });
+  const accessControlPreflightResponse = await preflightHelper.accessControlPreflightCheck(
+    accessControlPreflightObjects,
+    namespace
+  );
```

Comment:
@SiddharthMantri I am not thrilled for this to live here. I was hoping to better contain changes to SOR code within the security extension/access control service.